test/e2e/autoscaling_test.go

```diff
@@ -23,38 +24,63 @@ func TestAutoscaling(t *testing.T) {
 	t.Parallel()
 	g := NewWithT(t)
 
+	if globalOpts.configurableClusterOptions.NodePoolReplicas != 1 {
+		t.Fatalf("this test must be run with --e2e.node-pool-replicas=1, is configured to run with %d", globalOpts.configurableClusterOptions.NodePoolReplicas)
+	}
+	if len(globalOpts.configurableClusterOptions.Zone) == 0 {
+		t.Fatal("this test must be run with multiple Availability Zones")
+	}
+
 	client, err := e2eutil.GetClient()
 	g.Expect(err).NotTo(HaveOccurred(), "failed to get k8s client")
 
 	ctx, cancel := context.WithCancel(testContext)
 	defer cancel()
 
-	if len(globalOpts.configurableClusterOptions.Zone) == 0 {
-		t.Fatal("TestAutoscaling requires multiple zones")
-	}
-
 	clusterOpts := globalOpts.DefaultClusterOptions(t)
 
 	hostedCluster := e2eutil.CreateCluster(t, ctx, client, &clusterOpts, globalOpts.Platform, globalOpts.ArtifactDir)
 
 	// Get one of the newly created NodePools
-	zone := globalOpts.configurableClusterOptions.Zone[0]
-	nodepool := &hyperv1.NodePool{
-		ObjectMeta: metav1.ObjectMeta{
-			Namespace: hostedCluster.Namespace,
-			Name:      e2eutil.NodePoolName(hostedCluster.Name, zone),
-		},
+	nodepools := &hyperv1.NodePoolList{}
+	if err := client.List(ctx, nodepools, crclient.InNamespace(hostedCluster.Namespace)); err != nil {
+		t.Fatalf("failed to list nodepools in namespace %s: %v", hostedCluster.Namespace, err)
```

Comment:
note: there might be acceptable transient issues which makes this call fail and the test should still pass. We should probably wrap all api server calls. Though this is to be discussed in a different PR.

Reply:
I agree, but we don't retry api calls in any test right now, should probably change that